src/hooks/useProposalNonVotes.ts

```diff
@@ -0,0 +1,33 @@
+import { useQuery } from "@tanstack/react-query";
+import { PaginatedResult } from "@/app/lib/pagination";
+import { fetchVotersWhoHaveNotVotedForProposal } from "@/app/proposals/actions";
+
+interface Props {
+  enabled: boolean;
+  limit?: number;
+  offset?: number;
+  proposalId: string;
+}
+
+const QK = "nonvotes";
+
+export const useProposalNonVotes = ({
+  enabled,
+  limit,
+  offset,
+  proposalId,
+}: Props) => {
+  const { data, isFetching, isFetched } = useQuery({
+    enabled: enabled,
+    queryKey: [QK, proposalId],
+    queryFn: async () => {
+      return (await fetchVotersWhoHaveNotVotedForProposal(proposalId, {
+        offset: offset || 0,
+        limit: limit || 20,
+      })) as PaginatedResult<any[]>;
+    },
+    staleTime: 180000, // 3 minute cache
```

Comment:
hmm does this mean if someone votes then they won't immediately see their name go away from the non-voter list? If they refresh it will wipe the cache though, so its not a huge deal? I don't think we are optimistically updating react-query cache on that mutation anyways so maybe not a big deal